spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/data/RandomData.java

```diff
@@ -329,6 +329,8 @@ public Object primitive(Type.PrimitiveType primitive) {
           return UTF8String.fromString((String) obj);
         case DECIMAL:
           return Decimal.apply((BigDecimal) obj);
+        case UUID:
+          return UTF8String.fromString(UUID.nameUUIDFromBytes((byte[]) obj).toString());
```

Comment:
Why does `generatePrimitive` provide `byte[]`? Shouldn't it create a String for Spark already?

Reply:
my guess would be because `RandomUtil.generatePrimitive(..)` is used in other places where UUIDs are expected to be `byte[]`